sycl/include/CL/sycl/ONEAPI/group_algorithm.hpp

```diff
@@ -204,19 +203,19 @@ template <typename Group> bool all_of(Group, bool pred) {
 }
 
 template <typename Group, typename T, class Predicate>
-bool all_of(Group g, T x, Predicate pred) {
-  static_assert(sycl::detail::is_generic_group<Group>::value,
-                "Group algorithms only support the sycl::group and "
-                "ONEAPI::sub_group class.");
+detail::enable_if_t<(detail::is_generic_group<Group>::value &&
+                     detail::is_arithmetic<T>::value),
```

Comment:
What is the reason for adding is_arithmetic extra restriction? Alike question is applicable for other functions.

Reply:
You're right.  Whatever `T` is, `pred(x)` is going to return a `bool`, and that's all that matters.  Checking `is_arithmetic` in this case is unnecessarily restrictive.

Fixed in 282634f7426200d35b8447afb5def2fa6b346876 .